core/templates/dev/head/domain/classifier/PythonProgramTokenizer.js

```diff
@@ -0,0 +1,377 @@
+      "u'''": single3prog, 'u"""': double3prog,
+      "ur'''": single3prog, 'ur"""': double3prog,
+      "R'''": single3prog, 'R"""': double3prog,
+      "U'''": single3prog, 'U"""': double3prog,
+      "uR'''": single3prog, 'uR"""': double3prog,
+      "Ur'''": single3prog, 'Ur"""': double3prog,
+      "UR'''": single3prog, 'UR"""': double3prog,
+      "b'''": single3prog, 'b"""': double3prog,
+      "br'''": single3prog, 'br"""': double3prog,
+      "B'''": single3prog, 'B"""': double3prog,
+      "bR'''": single3prog, 'bR"""': double3prog,
+      "Br'''": single3prog, 'Br"""': double3prog,
+      "BR'''": single3prog, 'BR"""': double3prog,
+      r: null, R: null, u: null, U: null,
+      b: null, B: null
+    };
+
+    var tripleQuoted = [
+      "'''", '"""', "r'''", 'r"""', "R'''", 'R"""',
+      "u'''", 'u"""', "U'''", 'U"""', "ur'''", 'ur"""', "Ur'''", 'Ur"""',
+      "uR'''", 'uR"""', "UR'''", 'UR"""', "b'''", 'b"""', "B'''", 'B"""',
+      "br'''", 'br"""', "Br'''", 'Br"""', "bR'''", 'bR"""', "BR'''", 'BR"""'];
+
+    var singleQuoted = [
+      "'", '"', "r'", 'r"', "R'", 'R"', "u'", 'u"', "U'", 'U"', "ur'",
+      'ur"', "Ur'", 'Ur"', "uR'", 'uR"', "UR'", 'UR"', "b'", 'b"', "B'", 'B"',
+      "br'", 'br"', "Br'", 'Br"', "bR'", 'bR"', "BR'", 'BR"' ];
+
+    var tabsize = 8;
+
+    return {
+      generateTokens: function(program) {
+        var tokenizedProgram = [];
+        var lnum = parenlev = continued = 0;
+        var namechars = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ_';
+        var numchars = '0123456789';
+        var contstr = '';
+        var needcont = 0;
+        var contline = null;
+        var indents = [0];
+        var lcount = 0;
+
+        while (1) {
+          var line = program[lcount];
+          lcount++;
+          if (line === undefined) {
+            break;
+          }
+          var pos = 0;
+          var max = line.length;
+
+          if (contstr) {
+            if (!line) {
+              // Exception.
+              $log.error('EOF in multi-line string');
+            }
+
+            endmatch = endprog.exec(line);
+            if (endmatch && endmatch.index === 0) {
+              token = endmatch[0];
+              pos = pos + token.length;
+              tokenizedProgram.push([PythonProgramTokenType.STRING, token]);
+              contstr = '';
+              needcont = 0;
+              contline = null;
+            }
+
+            else if (
+              needcont && line.slice(-2) !== '\\\n' ||
+              line.slice(-3) !== '\\\r\n') {
+              tokenizedProgram.push(
+                [PythonProgramTokenType.ERRORTOKEN, contstr + line]);
+              contstr = '';
+              contline = null;
+              continue;
+            }
+
+            else {
+              contstr = contstr + line;
+              contline = contline + line;
+              continue;
+            }
+          }
+
+          // new statement
```

Comment:
For consistency, capitalize and add a period (same applies for all other comments).